src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java

```diff
@@ -194,6 +196,41 @@ String getAppPath(String appName) {
         return appPath.toString() + "/" + appName;
     }
 
+    /**
+     * If we are not using JavaFX, we immediately return the provided classpath, no further processing needed
+     * If we use JavaFX, we will first obtain the location of the JavaFX SDK for this configuration.
+     * This may throw an IOException.
+     * After the path to the JavaFX SDK is obtained, the JavaFX jars for the host platform are replaced by
+     * the JavaFX jars for the target platform.
+     * @param classPath
+     * @return
+     * @throws IOException
+     */
+    @Override
+    String processClassPath(String classPath) throws IOException {
+        if (!this.projectConfiguration.isUseJavaFX()) {
+            return classPath;
+        }
+        // we are using JavaFX
+        String javafxSDK = FileDeps.getJavaFXSDK(projectConfiguration).resolve("lib").toString();
```

Comment:
We already have `projectConfiguration .getJavafxStaticLibsPath()`, but this new method basically calls `setupDependencies`. We were doing that on `nativeLink`, probably we should move it to `nativeCompile`?

Reply:
setupDependencies sounds very abstract and doesn't indicate what is actually going to happen.
The FileDeps.getJavaFXSDK is very concrete and clearly explains what it does. It will invoke setupDependencies only when needed, and the caller of this method clearly knows what is being returned.

Reply:
That's okay, but the problem is that we still have `projectConfiguration.getJavafxStaticLibsPath()`, that is calling from many different places.

I'd suggest having a "path validation" in FileDesps, like the one you are proposing for JavaFXSDK: every time you call a path, check in FileDeps if the path exists, else call setupDependencies, else throw exception.

Reply:
We can do this in a follow-up PR.